Create the view of group#edit and add group update function

tokuchan-pg opened the pull request:
# WHAT
グループ編集画面の作成と、グループ編集機能の実装

# WHY
グループ管理機能に必須のため

### 補足
グループ作成画面(new.html.haml)とグループ 編集画面(edit.html.haml)の共通部分は部分テンプレート化（_group_form.html.haml）

### グループ編集画面↓
![2017-04-24 21 33 43](https://cloud.githubusercontent.com/assets/25572309/25337245/1389da40-2936-11e7-8a59-005b5ff53382.png)

tokuchan-pg committed:
created the view of group#edit by copying the view of group#new

tokuchan-pg committed:
made partial template for group#new & group#edit view

tokuchan-pg committed:
defined group#edit & group#update action

miyagit commented on app/controllers/groups_controller.rb:
```diff
@@ -15,6 +15,21 @@ def create
+    @group = Group.find(params[:id])
+    if @group.update(group_params)
+      flash.notice = 'グループが更新されました'
+      redirect_to root_path
```
redirect_toのオプションでフラッシュメッセージを出すことができるので、
フラッシュメッセージはredirect_toのオプションで出すようにしてください。

tokuchan-pg replied:
createアクションとupdateアクションにて対応いたしました。
```
redirect_to root_path, notice: '新規グループが作成されました。'
```
```
redirect_to root_path, notice: 'グループが更新されました'
```

miyagit commented on app/views/groups/_group_form.html.haml:
```diff
@@ -0,0 +1,30 @@
+=form_for @group do |f|
+  .chat-group-form__field
+    .chat-group-form__field--left
+      %label.chat-group-form__label{:for => "chat_group_name"} グループ名
```
ハッシュロケットは使わないようにしましょう。

tokuchan-pg replied:
修正いたしました。

miyagit commented on app/views/groups/edit.html.haml:
```diff
@@ -0,0 +1,3 @@
+.chat-group-form
+  %h1 チャットグループ編集
```
%h1 チャットグループ編集の記述は消して、部分テンプレート先の条件式で
%h1 チャットグループ編集を出力するようにしましょう。

tokuchan-pg replied:
修正いたしました。

miyagit commented on app/views/groups/new.html.haml:
```diff
@@ -1,33 +1,3 @@
 .chat-group-form
   %h1 新規チャットグループ
```
こちらも同じく%h1 新規チャットグループの記述は消して、
部分テンプレート先の条件式で出力するようにしましょう。

またその条件式はヘルパーメソッドを定義して、そのヘルパーメソッドを部分テンプレート先で呼ぶことで
%h1 新規チャットグループを出力する条件式の実装をしましょう。

tokuchan-pg replied:
修正いたしました。

tokuchan-pg committed:
show flash message as a option of redirect_to method

tokuchan-pg committed:
replaced hash rocket

tokuchan-pg committed:
updated the way of rendering partial template

Used ‘current_page?’ in helpers/group_helper.rb to change the text to
show in h1 tag based on the action called.

tokuchan-pg commented:
部分テンプレートについて、主に以下の修正を加えました。

- h1タグも部分テンプレート内に記述する

- helpers/group_helper.rbにヘルパーメソッドを定義。メソッドの中で、current_page?を使うことでリクエストが生成されたアクションを判別し、h1タグに出力する文字列が切り替わるようにする

- 部分テンプレート内でヘルパーメソッドを呼び出す

tokuchan-pg committed:
revised the markup structure

there was a mistake in former commit…

motemotenten commented on app/helpers/group_helper.rb:
```diff
@@ -0,0 +1,5 @@
+module GroupHelper
+  def render_partial_template
+    current_page?(action: :new)? '新規チャットグループ' : 'チャットグループ編集'
```
三項演算子使えてるのいいですね

tokuchan-pg replied:
ありがとうございますm(_ _)m

motemotenten commented on app/controllers/groups_controller.rb:
```diff
@@ -7,14 +7,27 @@ def new
+  end
+
+  def update
+    @group = Group.find(params[:id])
```
こちらの記述重複しているので、まとめましょう。

tokuchan-pg replied:
private以下にインスタンス変数をセットするメソッドを定義して、それをbefore_actionで呼び出す形でまとめました。

tokuchan-pg committed:
Method codes refactoring

By defining a private method and call it by before_action
